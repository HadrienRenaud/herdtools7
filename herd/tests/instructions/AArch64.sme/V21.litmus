AArch64 V21
{
  uint32_t x[4];
  uint32_t y[4];
  0:X0=x; 0:X1=y;
}
P0 ;
SMSTART                      ;
INDEX Z0.S,#1,#1             ;
PTRUE P0.S                   ;
PTRUE P1.S,VL2               ;
ADDVA ZA0.S,P0/M,P1/M,Z0.S   ;
PTRUE P2.S,VL4               ;
ST1W {ZA0H.S[W12,0]},P2,[X0] ;
ST1W {ZA0V.S[W12,0]},P2,[X1] ;
SMSTOP                       ;

forall x={1,1,0,0} /\ y={1,2,3,4}
