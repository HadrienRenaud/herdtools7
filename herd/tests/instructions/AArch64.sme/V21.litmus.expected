Test V21 Required
States 1
x={1,1,0,0}; y={1,2,3,4};
Ok
Witnesses
Positive: 1 Negative: 0
Flag Scalable-Matrix-Extensions-is-work-in-progress
Condition forall (x={1,1,0,0} /\ y={1,2,3,4})
Observation V21 Always 1 0
Hash=e488225ea924f48a536cd42e3c785833

